libs/register/feature/src/lib/register.component.ts

```diff
@@ -10,34 +14,356 @@ export class RegisterComponent {
+      }
+    }
+
+    dom = document.getElementById("snameError");
+    if(!this.parentRegisterDetailsForm.controls['surname'].valid)
+    {
+      formError = true
+      if(dom != null)
+      {
+        dom.innerHTML = "Surname is empty";
+        dom.style.display = "block";
+      }
+    }
+    else
+    {
+      if(dom != null)
+      {
+        dom.style.display = "none";
+      }
+    }
+
+    dom = document.getElementById("emailError");
+    if(!this.parentRegisterDetailsForm.controls['email'].valid)
+    {
+      formError = true
+      if(dom != null)
+      {
+        dom.innerHTML = "Invalid email";
+        dom.style.display = "block";
+      }
+    }
+    else
+    {
+      if(dom != null)
+      {
+        dom.style.display = "none";
+      }
+    }
+
+    dom = document.getElementById("numberError");
+    if(!this.parentRegisterDetailsForm.controls['phone'].valid)
+    {
+      formError = true
+      if(dom != null)
+      {
+        dom.innerHTML = "Invalid phone number";
+        dom.style.display = "block";
+      }
+    }
+    else
+    {
+      if(dom != null)
+      {
+        dom.style.display = "none";
+      }
+    }
+
+    dom = document.getElementById("idError");
+    if(!this.parentRegisterDetailsForm.controls['id'].valid)
+    {
+      formError = true
+      if(dom != null)
+      {
+        dom.innerHTML = "Invalid ID";
+        dom.style.display = "block";
+      }
+    }
+    else
+    {
+      if(dom != null)
+      {
+        dom.style.display = "none";
+      }
+    }
+
+    if(this.parentChosen)
+    {
+      dom = document.getElementById("medError");
+      if(!this.parentRegisterDetailsForm.controls['medAid'].valid)
+      {
+        formError = true
+        if(dom != null)
+        {
+          dom.innerHTML = "Invalid medical aid number";
+          dom.style.display = "block";
+        }
+      }
+      else
+      {
+        if(dom != null)
+        {
+          dom.style.display = "none";
+        }
+      }
+    }
 
-    if(this.parentRegisterDetailsForm.get('pass')?.value === this.parentRegisterDetailsForm.get('confPass')?.value) {
+    dom = document.getElementById("addrError");
+    if(dom != null)
+    {
+      this.getLocations();
+      if (this.potentialLocations.indexOf(this.parentRegisterDetailsForm.value.address) == -1)
+      {
+        formError = true
+        this.locationError = true;
+        dom.innerHTML = "Please select a valid location from the suggested below.";
+        dom.style.display = "block";
+      }
+      else
+      {
+        this.locationError = false;
+        if(dom != null)
+        {
+          dom.style.display = "none";
+        }
+      }
+    }
+
+    dom = document.getElementById("pswError");
+    if(!this.parentRegisterDetailsForm.controls['pass'].valid)
+    {
+      formError = true
+      if(dom != null)
+      {
+        dom.innerHTML = "Invalid password : should contain upper and lowercase characters aswell as special characters and numbers";
```

Comment:
aswell -> as well

Also please check consistency with other errors like ending some errors with full stops and others not?